CloudSQLReplicator/google/cloud/pontem/sql/replicator/util/gcp_api_util.py

```diff
@@ -88,10 +93,91 @@ def build_sql_admin_service(credentials=None):
         credentials or default_credentials
     )
 
+    service = discovery.build(
+        service,
+        version,
+        http=authorized_http
+    )
+
+    return service
+
+# Compute
+def build_compute_service(credentials=None):
+    """Authorizes and sets custom CloudSQL Replicator user agent.
+
+    Args:
+        credentials (google.auth.Credentials): Credentials to authorize client.
+
+    Returns:
+        Resource: Authorized compute service proxy with custom user agent.
+    """
+    service = build_authorized_svc(
+        COMPUTE_SVC,
+        COMPUTE_SVC_VERSION,
+        credentials
+    )
+
+    return service
+
+
+def create_firewall_rule(name, source_ip_range, protocol='TCP',
+                         is_allowed=True, ports=frozenset(['3306']),
+                         is_ingress=True,
+                         project=None, credentials=None):
+    """Creates a firewall rule in a Google Cloud Project.
+
+    Args:
+        name (str): name of resource, must comply with RFC1035.
+        source_ip_range (list): list of source IP ranges in CIDR format.
+        protocol (str): protocol allowed.
+        is_allowed (bool): whether traffic allowed or denied.
+        ports (str): port for traffic.
+        is_ingress (bool): whether traffic is ingress or egress.
+        project (str): Project ID where replica will be created.
+        credentials (google.auth.Credentials): Credentials to authorize client.
+
+    Returns:
+        JSON: Response from request.
+
+    Raises:
+        ValueError: Exception if name does not conform to RFC1035.
+    """
+    if re.match(RFC1035_REGEX, name) is None:
+        raise ValueError('{} does not conform to RFC1035'.format(name))
+    default_credentials, default_project = google.auth.default()
+    service = build_compute_service(credentials or default_credentials)
+    request_body = {
+        'name': name,
+        'sourceRanges': source_ip_range,
+        'allowed' if is_allowed else 'denied': {
+            'IPProtocol': protocol,
+            'ports': ports,
+        },
+        'direction': 'INGRESS' if is_ingress else 'EGRESS',
```

Comment:
Isn't there a chance that direction could be both?

Reply:
The API expects one of ["INGRESS", "EGRESS"].